### PR TITLE
21534 alt response codes

### PR DESCRIPTION
**Pull request by brectanus-sigsci**
This reorganizes the module config to a central `ModuleConfig` so that the config is easier to share among internal modules like envoy and revproxy. In addition, this adds an `AltResponseCodes` configuration and associated logic/tests to use it in the Module wrapper.

### CODE REVIEW

**Review by nickg-sigsci — APPROVED** [review]
great

File: module.go [hunk 1]
```diff
@@ -18,82 +15,61 @@ import (
...
-	inspector                 Inspector
-	inspInit                  InspectorInitFunc
-	inspFini                  InspectorFiniFunc
-	headerExtractor           func(*http.Request) (http.Header, error)
+	config          *ModuleConfig
+	handler         http.Handler
+	inspector       Inspector
+	inspInit        InspectorInitFunc
+	inspFini        InspectorFiniFunc
+	headerExtractor HeaderExtractorFunc
```
Comment (mlakes-sigsci) [inline module.go:27]: Isnt headerExtractor already in the config object(ModuleConfig). Repeated here? Same for inspFini, et al?

**Review by mlakes-sigsci — COMMENTED** [review]
Comments within...Comments within..duplication of config fields between ModuleConfig and Module?

File: module.go (hunk 1 above)
Comment (brectanus-sigsci) [reply module.go:27]: Yes, but it is done this way as it is a bit more efficient.

**Review by mlakes-sigsci — APPROVED** [review]
lgtm

File: module.go [hunk 2]
```diff
@@ -271,11 +120,11 @@ func (m *Module) ServeHTTP(w http.ResponseWriter, req *http.Request) {
...
 
 	wafresponse := out.WAFResponse
-	switch wafresponse {
-	case 406:
+	switch {
+	case m.config.IsBlockCode(int(wafresponse)):
 		status := int(wafresponse)
 		http.Error(rw, fmt.Sprintf("%d %s\n", status, http.StatusText(status)), status)
-	case 200:
+	case m.config.IsAllowCode(int(wafresponse)):
```
Comment (sada-sigsci) [inline module.go:124]: I think first case should be `IsAllowCode` so it is more efficient

**Pull request merged**